src/main/java/frc/utils/Translation2dUtils.java

```diff
@@ -0,0 +1,26 @@
+package frc.utils;
+
+import edu.wpi.first.math.geometry.Translation2d;
+
+public class Translation2dUtils {
+
+
+	public static Translation2d getClosestPoint(Translation2d origin, Translation2d... points) {
+		Translation2d closest = new Translation2d(Double.MAX_VALUE, Double.MAX_VALUE);
+
+		for (Translation2d point : points) {
+			if (point != null) {
+				if (point.getDistance(origin) < closest.getDistance(origin)) {
+					closest = point;
+				}
+			}
+		}
+		return closest;
+	}
```

Comment:
the origin could be closer to the one with the Double.MAX_VALUE than the other points, and maybe make it so it returns an Optional that would not contain any value if the closest translation is the MAX_VALUE one.

Reply:
could make points[0] default

Reply:
point[0] could be null

Reply:
@hillelv @GilStein1

Reply:
Double.MAX_VALUE is not infinity. if origin is a result of some division by a number that's close to 0 than it will be closer to the Double.MAX_VALUE. If that's not the case with origin then it's fine but I don't like the varargs thing since it throws an exception

Reply:
if I understand correctly,
if origin is a division closer to 0 (like 0.00001), any other double will be closer to Double.MAX_VALUE
if that is the case, you are not correct, as you can see in this code:
![image](https://github.com/user-attachments/assets/86ec6c22-c80c-41e6-bd1d-543e255a0492)
or in this code:
![image](https://github.com/user-attachments/assets/744bf207-9818-4be2-be0a-ca860c588843)